src/db/index.ts

```diff
@@ -53,6 +53,32 @@ export class FrappeDB {
       });
   }
 
+  /**
+   * Get a document from the database
+   * @param {string} doctype Name of the doctype
+   * @param {@type GetDocArgs} [args] Arguments for the query
+   * @returns Promise which resolves to the document object
+   */
+  async getLastDoc<T>(doctype: string, args?: GetLastDocArgs): Promise<FrappeDoc<T>> {
+
+    if (!args || !args?.orderBy) {
+      args = {
+        ...args,
+        orderBy: {
+          field: 'creation',
+          order: 'desc'
+        }
+      }
+    }
+
+    const getDocLists = await this.getDocList<T & { name?: string }>(doctype, { ...args, limit: 1 });
```

Comment:
```suggestion
    const getDocLists = await this.getDocList<{ name: string }>(doctype, { ...args, limit: 1 });
```

The `getDocList` function will only fetch list of names (primary keys) - not `<T>` since that is for the entire document.